### PR TITLE
test: Fix role names in OpenShift marmalade test project; adjust to deprecation of -v with multiple values

### PR DESCRIPTION
**Pull request by martinpitt**
It is very confusing to pre-create the "amanda" and other users in our
OpenShift test VM:

    # oc get rolebindings -n marmalade
    NAME      ROLE      USERS                  GROUPS    SERVICE ACCOUNTS   SUBJECTS
    admin     /admin    scruffy
    edit      /edit     scruffy, amanda
    view      /view     scruffy, tom, amanda   sports

But these two roles aren't actually valid in OpenShift:

    # oadm policy add-role-to-user edit amanda -n marmalade
    The RoleBinding "edit" is invalid.

This causes these users/groups to be shown on the "Show all projects"
Users list and shown as being members of marmalade, but are not visible
on the marmalade project page (after actually creating the user by
logging in as them).

Adjust the "oc patch" command to create these roles with the official
registry-{editor,viewer} roles so that they actually work. This avoids
confusion for people who use our VM to test the cockpit registry.
Notably none of our tests actually rely on these two users/roles, these
are just for human testing.

### CODE REVIEW

**Comment by martinpitt** [comment]
bot: Image refresh for openshift

**Comment by cockpituous** [comment]
Image creation for openshift in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-openshift-732a52a30166e06ccf6c9f5959273a952b7eb564/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-openshift-732a52a30166e06ccf6c9f5959273a952b7eb564

**Comment by martinpitt** [comment]
I get a local failure (with the new image), need to investigate.

**Comment by martinpitt** [comment]
This seems to expose a bug in changing project policy when the roles already exist:
```
$ test/verify/check-openshift TestRegistry.testProjectAdmin
> RoleBinding "registry-viewer" is invalid: metadata.resourceVersion: Invalid value: "": must be specified for an update
```
(This is also being shown in the dialog box). I'm looking into that.

**Comment by martinpitt** [comment]
OK, figured it out by diffing `oc export policybinding -n marmalade` before/after when manually configuring the roles through cockpit's registry. Our initial json dump needs to include the `resourceVersion` as any change expects that to already exist. Rebuilding image, keeping needswork until I push the UUID refresh.

**Comment by martinpitt** [comment]
bot: Image refresh for openshift

**Comment by cockpituous** [comment]
Image creation for openshift in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-openshift-aa77e383dea6d5ff1000e1c0b36cbe0aed173494/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-openshift-aa77e383dea6d5ff1000e1c0b36cbe0aed173494

**Comment by martinpitt** [comment]
Re-pushed with updating test/verify/files/openshift.kubeconfig from the updated image.

**Comment by dperpeet** [comment]
container test failures seem relevant:
```
error: error processing the template "cockpit-openshift-template": Template "cockpit-openshift-template" is invalid: template.parameters[0]: Required value: template.parameters[0]: parameter OPENSHIFT_OAUTH_PROVIDER_URL is required and must be specified
Wrote TestOpenshift-testBasic-FAIL.png
Journal extracted to TestOpenshift-testBasic-10.111.112.101-FAIL.log
not ok 8 testBasic (check_kubernetes_openshift.TestOpenshift) duration: 19s
Traceback (most recent call last):
  File "/build/cockpit/test/verify/kubelib.py", line 639, in testBasic
    self.login_and_go("/kubernetes")
  File "/build/cockpit/test/common/testlib.py", line 537, in login_and_go
    self.machine.start_cockpit(host)
  File "/build/cockpit/test/containers/check-kubernetes-openshift", line 99, in start_cockpit
    self.execute("oc process -f /tmp/cockpit-openshift-template.json -v {0} > out.json".format(','.join(args)))
  File "/build/cockpit/test/common/testvm.py", line 412, in execute
    raise subprocess.CalledProcessError(proc.returncode, command, output=output)
CalledProcessError: Command 'oc process -f /tmp/cockpit-openshift-template.json -v COCKPIT_KUBE_INSECURE=true,COCKPIT_KUBE_URL=http://10.111.112.101:30000,OPENSHIFT_OAUTH_PROVIDER_URL=https://10.111.112.101:8443 > out.json' returned non-zero exit status 1

```

**Comment by martinpitt** [comment]
Right, they are. I don't know yet how to reproduce them locally, that'll be one of my next exercises :-) (This also isn't urgent, it just created quite some confusion while I was working on a "real" bug).

**Comment by martinpitt** [comment]
With the new OpenShift version in the image rebuild the old comma-separated multiple values with `-v` syntax stopped working, so adjust documentation and tests accordingly. I also added some `test/README` documentation to explain how to run the `test/container` tests locally.

File: test/common/testlib.py [hunk 1]
```diff
@@ -534,7 +534,7 @@ def tearDown(self):
         shutil.rmtree(self.tmpdir)
 
     def login_and_go(self, path=None, user=None, host=None, authorized=True):
-        self.machine.start_cockpit(host)
```
Comment (dperpeet) [inline test/common/testlib.py]: this is passed to https://github.com/cockpit-project/cockpit/blob/919f083e40b3000549a62d08fcbd7d3aed2a4e5e/test/common/testvm.py#L570 in our tests, I don't think we should remove it at this point

Comment (martinpitt) [inline test/common/testlib.py]: Argh, brown paperbag, thanks for catching. Pushed a fix.

**Comment by dperpeet** [comment]
I assume this was just a race on fedora-25?
```
# testProjectUsers (check_openshift.TestRegistry)
#
No resources found.
Wrote TestRegistry-testProjectUsers-FAIL.png
Journal extracted to TestRegistry-testProjectUsers-10.111.114.144-FAIL.log
Journal extracted to TestRegistry-testProjectUsers-10.111.112.101-FAIL.log
not ok 70 testProjectUsers (check_openshift.TestRegistry) duration: 192s
Traceback (most recent call last):
  File "./verify/check-openshift", line 643, in testProjectUsers
    wait(lambda: username not in o.execute("oc get rolebinding -n testprojectuserproj"))
  File "/build/cockpit/test/common/testlib.py", line 1154, in wait
    raise Error(msg or "Condition did not become true.")
Error: Condition did not become true.
```

**Comment by martinpitt** [comment]
@dperpeet: Yes, it was an unrelated fluke. It's all green now. This unblocks the refreshing of openshift images, so it would be nice to land this soon.

**Pull request closed without merging**